File: SDKs/googleapis.com/jobs/v3p1beta1/java/lib/src/main/java/openapisdk/models/shared/RequestMetadata.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * RequestMetadata
 * Input only. Meta information related to the job searcher or entity conducting the job search. This information is used to improve the performance of the service.
**/
public class RequestMetadata {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("deviceInfo")
    public DeviceInfo deviceInfo;
    public RequestMetadata withDeviceInfo(DeviceInfo deviceInfo) {
        this.deviceInfo = deviceInfo;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("domain")
    public String domain;
    public RequestMetadata withDomain(String domain) {
        this.domain = domain;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("sessionId")
    public String sessionId;
    public RequestMetadata withSessionId(String sessionId) {
        this.sessionId = sessionId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("userId")
    public String userId;
    public RequestMetadata withUserId(String userId) {
        this.userId = userId;
        return this;
    }
}

File: SDKs/googleapis.com/jobs/v3p1beta1/python/src/sdk/models/shared/requestmetadata.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import deviceinfo as shared_deviceinfo


@dataclass_json
@dataclasses.dataclass
class RequestMetadata:
    r"""RequestMetadata
    Input only. Meta information related to the job searcher or entity conducting the job search. This information is used to improve the performance of the service.
    """
    
    device_info: Optional[shared_deviceinfo.DeviceInfo] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('deviceInfo') }})
    domain: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('domain') }})
    session_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sessionId') }})
    user_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('userId') }})
    
